### DISCUSSION TITLE
Comment ouvrir un modal présent dans une scène A depuis une scène B

### DISCUSSION THREAD

**Discussion by JeremieDENAEGEL** [witsa/synapps#685] [Q&A]
Bonjour,

Je voudrais ouvrir un modal présent dans une sceneA depuis une sceneB où se trouve mon acteur Bouton.

Dans le script OnClick de l'acteur Bouton de la sceneB, par quoi faut-il changer "context.getActor" pour atteindre la sceneA ? 
`context.getActor('modal_InfoSonde').properties.isShown = true;`

Merci d'avance ;)

**Comment by manufitoussiwit** [comment]
Par script, on peut accéder à la scène racine ou la scène dans laquelle l'acteur écran présente la scène courante. C'est pas idéal d'aller chercher un acteur comme ça. Mais c'est possible.

https://witsa.github.io/synapps/script-api/Synapps.Scene.html#screen
![image](https://github.com/witsa/synapps/assets/35595723/4befc495-e06b-42ad-9d53-2aa61c3d3dc8)

La scène courante : `context.stage` (ou `context.scene`) a la propriété `screen` qui est l'acteur écran qui la montre ou `null` si la scène est montrée directement.

Par conséquent, avec `context.scene.screen`, on peut atteindre la scène de l'acteur écran : `context.scene.screen.scene`. 
On peut faire appelle ensuite à la fonction `getActor` pour aller chercher la modale : `context.scene.screen.scene.getActor('modal_InfoSonde')`. 
Il faudra veiller au cas où l'acteur modale n'est pas trouvé avant de l'utiliser pour éviter de générer une erreur.

```javascript

if (context.scene.screen) {
  const modal = context.scene.screen.scene.getActor('modal_InfoSonde');
  if (modal) {
    modal.properties.isShown = true;
  }
}

```

**Reply by manufitoussiwit** [reply]
Il faut dire que c'est pas une super manip. Une autre méthode serait de taper sur une additionnel sur la scène principale qui est liée à la propriété `isShown `de la modale. Comme ça on pas besoin de connaitre la clé de l'acteur modale dans la sous scène.
